plutus-ir/src/Language/PlutusIR.hs

```diff
@@ -118,18 +117,17 @@ data Term tyname name a =
 
 instance (Serialise a, Serialise (tyname a), Serialise (name a)) => Serialise (Term tyname name a)
 
-embedIntoIR :: PLC.Term tyname name a -> Term tyname name a
```

Comment:
I'm afraid it's not quite that easy - you'll need something like `embed :: TermLike term tyname name => Term tyname name a -> term tyname name a`

Reply:
The logic was that "anything that we might need `embedIntoIR` for could as well just return a `TermLike` directly - I think this can always be done (there are only 2 uses of `embedIntoIR` in the codebase), but I'm not entirely confident.

Reply:
I'm pretty sure this is not true or at least inconvenient.